test/test_infer.rb

```diff
@@ -92,7 +148,7 @@ def return_two_plus_two
   def plus_two(val)
     val + 2
   end
-  should_have_type :plus_two, '([ +: (Number) -> aaa ]) -> aaa'
+  should_have_type :plus_two, '([ +: (Number) -> a ]) -> b'
```

Comment:
This is going to fail with a really strange error. It will say:

```
Expected ([ +: (Number) -> a ]) -> b
Got      ([ +: (Number) -> { { TestInfer#plus_two arg: val }#+ ret: ret } ]) -> { TestInfer#plus_two ret: ret }
```

Which makes it seem like the type variables are the problem. However, changing the expected type to `([ +: (Integer) -> a ]) -> b` will cause it to pass. I'm not entirely sure what's going on, except that the internal representation of the solution is `Integer`, while the printed solution is `Number`.